core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/LogicalAndIf.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.batch.aggregator.function;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+
+import java.io.Serializable;
+
+/**
+ * Returns true if all the values in the group fulfill the condition given, false even if there is a single value that
+ * does not fulfill the condition. For example condition = value > 1
+ */
+public class LogicalAndIf extends LogicalAnd implements Serializable {
+  private final Condition condition;
+
+  public LogicalAndIf(String fieldName, Condition condition) {
+    super(fieldName);
+    this.condition = condition;
+  }
+
+  @Override
+  public void mergeValue(StructuredRecord record) {
+    if (!meetCondition(record, condition)) {
+      return;
+    }
+    super.mergeValue(record);
+  }
```

Comment:
Condition acts more like a filter for records, since the records are filtered out during mergeValue, we did not see the need to add any additional filter mergeAggregates

Reply:
We think this scenario is not possible because aggregates are grouped but by their aliases and we can not have multiple aggregates with the same alias. Below we can see that aggregates are grouped by their aliases.
https://github.com/cdapio/hydrator-plugins/blob/44065a6fa6121a86ea7bb5c7c2de3c3f10c7d28f/core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/GroupByAggregator.java#L371-L378

Also below we can see that they are merged based by key which we can see from above that key is the alias.
https://github.com/cdapio/hydrator-plugins/blob/44065a6fa6121a86ea7bb5c7c2de3c3f10c7d28f/core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/GroupByAggregator.java#L343-L345

Reply:
I see, thanks!